public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -146,6 +143,24 @@ export class LokiDatasource
     });
   }
 
+  // sometimes we need to run a loki query where we only have a LokiQuery object
+  _querySingle(query: LokiQuery, range: TimeRange, app: CoreApp, requestId: string): Observable<DataQueryResponse> {
```

Comment:
I am thinking if this couldn't be utility function that creates data request from loki query instead of creating AND running of query.

Reply:
done